packages/playground/website/playwright/e2e/website-ui.spec.ts

```diff
@@ -0,0 +1,175 @@
+import { test, expect } from '../playground-fixtures.ts';
+import {
+	clickCreateInNewSiteModal,
+	clickSaveInEditSettings,
+	expandSiteView,
+	getSiteInfoRowValue,
+	getSiteTitle,
+	hasNetworkingEnabled,
+	openEditSettings,
+	openNewSiteModal,
+	selectPHPVersion,
+	selectWordPressVersion,
+	setNetworkingEnabled,
+} from './website-helpers.ts';
+import { Blueprint } from '@wp-playground/blueprints';
+
+// We can't import the SupportedPHPVersions versions directly from the remote package
+// because of ESModules vs CommonJS incompatibilities. Let's just import the
+// JSON file directly. @ts-ignore
+// eslint-disable-next-line @nx/enforce-module-boundaries
+import { SupportedPHPVersions } from '../../../../php-wasm/universal/src/lib/supported-php-versions.ts';
+// eslint-disable-next-line @nx/enforce-module-boundaries
+import * as MinifiedWordPressVersions from '../../../wordpress-builds/src/wordpress/wp-versions.json';
+
+test('should reflect the URL update from the navigation bar in the WordPress site', async ({
+	page,
+}) => {
+	await page.goto('./?url=/wp-admin');
+
+	await expandSiteView(page);
+
+	const inputElement = page.locator('input[value="/wp-admin/"]');
+	await expect(inputElement).toBeVisible();
+	await expect(inputElement).toHaveValue('/wp-admin/');
+});
+
+test('should switch between sites', async ({ page }) => {
+	await page.goto('./');
+
+	await openNewSiteModal(page);
+
+	const newSiteName = await page
+		.locator('input[placeholder="Playground name"]')
+		.inputValue();
+
+	await clickCreateInNewSiteModal(page);
+
+	await expect(await getSiteTitle(page)).toMatch(newSiteName);
+
+	const sidebarItem = page.locator(
+		'.components-button[class*="_sidebar-item"]:not([class*="_sidebar-item--selected_"])'
+	);
+	const siteName = await sidebarItem
+		.locator('.components-flex-item[class*="_sidebar-item--site-name"]')
+		.innerText();
+	await sidebarItem.isVisible();
+	await sidebarItem.click();
+
+	await expect(await getSiteTitle(page)).toMatch(siteName);
```

Comment:
This assertion will be evaluated once, and if not matched, it will fail the test. To prevent this assertion from being flaky, we need to use the auto-waiting locator assertion [`toHaveText()`](https://playwright.dev/docs/api/class-locatorassertions#locator-assertions-to-have-text) instead:

```suggestion
	const siteName = page.locator('.components-flex-item[class*="_sidebar-item--site-name"]');
	await expect(siteName).toHaveText(await getSiteTitle(page));
```

Reply:
I ended up using `toMatch` here because I need to save the text before clicking.
After clicking the text changes.

Reply:
Looking at this again, this test might be producing false negative results. If we compare an element to another element, and both of them produce asynchronous content, we might end up in a situation where both elements are equally broken, but the test still passes. Can we be more deterministic here to increase the value of such a test?

For example:

```js
const expectedTitle = 'Groundplay';

await expect(/* <siteNameLocator> */).toHaveText(expectedTitle);
await expect(/* <siteTitleLocator> */).toHaveText(expectedTitle);
```

Reply:
This won't work in my case. 

The site title is a randomly generated name, this is why I'm first getting the title of the site from the button.
Then I use that title string in the matching.

The test doesn't compare elements.